### ISSUE TITLE
use timeout props without exit time

### ISSUE THREAD

**Issue by LiuuY** [reactjs/react-transition-group#460]
**Do you want to request a *feature* or report a *bug*?**

bug, maybe?

**What is the current behavior?**

if I use timeout props without exit time, eg, 
```javascript
<CSSTransition
  key={id}
  timeout={{
    enter: 300,
    // no exit time
  }}
  classNames="fade"
>
```

https://codesandbox.io/s/wnjp84n125

Click the red X button, after the animation, the element that should (?) be removed from DOM tree is still there.

Is this a feature or a bug?

![image](https://user-images.githubusercontent.com/14286374/53026910-d71d2780-349e-11e9-9a43-abb8f3b7e389.png)

version: 2.2.1

**Comment by silvenon** [comment]
I think this is unexpected behavior. I think the expected behavior should be to raise an error or warning, because I don't know if there is a sensible default value we could add for `exit`. (Maybe `0`?)

**Comment by silvenon** [comment]
Solved with #464. `exit` timeout now defaults to `0`, which I consider intuitive behavior.
